src/components/map/rent-popup.tsx

```diff
@@ -235,6 +245,8 @@ const RentPopup: React.FC<RentPopupProps> = ({
                     const handleClick = () =>
                       (!isReserved || isReservedByMe) && setSelectedSlot(slot);
 
+                    if (isReservedByMe && !selectedSlot) { setSelectedSlot(slot); }
```

Comment:
No imperative code in render functions please. This needs to be done differently. One way would be to extract the logic into a real react component and use an effect hook in that.

Reply:
Since you're checking for the selected slot this will not lead to an infinite rendering-loop, but calling setState within `render`-functions is strongly discouraged.

Reply:
See 3378a5ff38ef49219b1495dd3eddb392fadfbe88.
I'm not to happy with how the code looks though...